pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -947,30 +950,67 @@ func (b realByteSlice) Sub(start, end int) index.ByteSlice {
 // found entry. If f returns an error it stops decoding and returns the received error.
 // TODO: use the upstreamed version from Prometheus tsdb/index package when available.
 // TODO: see https://github.com/prometheus/prometheus/pull/11535
-func readPostingsOffsetTable(bs index.ByteSlice, off uint64, f func(labels.Label, uint64, int) error) error {
+func readOffsetTable[T any](bs index.ByteSlice, off uint64, read func(d *encoding.Decbuf) (T, error), f func(T, uint64, int) error) error {
```

Comment:
Switched back to the generic version of this which is the same as one I'm upstreaming in https://github.com/prometheus/prometheus/pull/11535